### ISSUE TITLE
VariableDeclarationUsageDistanceCheck doesn't handle method definition properly

### ISSUE THREAD

**Issue by jxr98** [checkstyle/checkstyle#13038]
I have read check documentation: https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below.

```bash
/var/tmp $ javac Test1.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <property name="charset" value="UTF-8"/>

    <!-- do not change severity to 'error', as that will hide errors caused by exceptions -->
    <property name="severity" value="warning"/>

    <!-- haltOnException is required for exception fixes and reporting of all exceptions -->
    <property name="haltOnException" value="false"/>

    <!-- BeforeExecutionFileFilters is required for sources of java9 -->
    <module name="BeforeExecutionExclusionFileFilter">
        <property name="fileNamePattern" value="module\-info\.java$" />
    </module>

    <module name="TreeWalker">
         <!-- Example of checkstyle Check usage -->
         <!-- PLEASE CHANGE IT TO CHECK YOU ARE TESTING !!!! -->
              
        <module name="VariableDeclarationUsageDistance">
          <property name="id" value="Custom"/>
          <property name="validateBetweenScopes" value="true"/>
          <property name="allowedDistance" value="1"/>
        </module> 
    </module>
    <!-- Example of filter -->
    <!--
    <module name="SeverityMatchFilter">
        <property name="severity" value="warning"/>
        <property name="acceptOnMatch" value="false"/>
    </module>
    -->

    <!-- as we run on regression even on non-compiled files we need to skip exceptions on them -->
    <module name="SuppressionSingleFilter">
      <property name="message" value="Exception occurred while parsing"/>
      <property name="checks" value="Checker"/>
    </module>

/var/tmp $ cat Test1.java

import java.util.Iterator;

public class Test1 {
    public void method(){
        int a=1;
        class GeneralLogic {
            public <T> Iterator<T> method(T[] b) throws Exception{
                if(a>0){
                    System.out.println(b.length);
                }
                return null;
            }
        }
    }
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.9.3-all.jar -c config.xml Test1.java

Starting audit...
[WARN] /var/tmp/Test.java:5:9: Distance between variable 'a' declaration and its first usage is 3, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [Custom]
Audit done.

```

For Windows users, please use `type` instead of `cat` and run

```cmd
set RUN_LOCALE="-Duser.language=en -Duser.country=US"
java %RUN_LOCALE% -jar checkstyle-10.9.3-all.jar -c config.xml Test1.java
```

in place of the last 2 commands above.

---

The distance should be 1. This three were caused by generic type which is `<T>`, parameter type in the method signature and `throws`. I think the correct distance one should be caused by the expression in the if condition.

**Comment by pymoura** [comment]
Hi, 
Is it okay to work on this issue? 
I'm not an experienced developer, but I would like to contribute to the project.

**Comment by jxr98** [comment]
> Hi,
> Is it okay to work on this issue?
> I'm not an experienced developer, but I would like to contribute to the project.

I think you can since I am not working on this issue. ☺️

**Comment by pymoura** [comment]
Great. I'm on it

**Comment by pymoura** [comment]
Hi @jxr98, 
I did some changes, and it seems it resolved this issue. However, I got into some error on another test that mentions issue #13011. 

 public void method9() {
        // until https://github.com/checkstyle/checkstyle/issues/13011
        Integer z = 5; // violation 'Distance .* is 3.'
        Iterator<Integer> mn = new HashSet<Integer>().iterator();
        class BClass extends Parent {
            Boolean h = false;
            @Override
            public <t> void xx(List<t> a) {
                if (a.get(0).toString().compareTo("1") > 0) {
                    System.out.println("temp");
                }
            }
Changing this line   Integer z = 5; // violation 'Distance .* is 2'. To report 2 instead of 3, all test pass.

Is the goal of this is to eventually have no violation after resolving issue#13011? Is it okay to make this change until 13011 gets resolved?

**Comment by romani** [comment]
almost done update is at https://github.com/checkstyle/checkstyle/pull/13065 , everyone in web is welcome to reuse and finish

**Comment by jxr98** [comment]
I'm on it. Will try to solve it within this week. Thanks. 😃

**Comment by Z4ID-Reaper** [comment]
I have read check documentation: https://checkstyle.org/config_coding.html#VariableDeclarationUsageDistance
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below.

@romani @Zopsss 
this issue should be included in "google style lable"

### config.xml
same as the user above.
```
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name = "Checker">
    <module name="TreeWalker">
        <module name="VariableDeclarationUsageDistance">
          <property name="validateBetweenScopes" value="true"/>
          <property name="allowedDistance" value="1"/>
        </module> 
    </module>
</module>
```

### Test.java
also same as the user above.
```
import java.util.Iterator;

public class Test1 {
    public void method(){
        int a=1;
        class GeneralLogic {
            public <T> Iterator<T> method(T[] b) throws Exception{
                if(a>0){
                    System.out.println(b.length);
                }
                return null;
            }
        }
    }
}
```

### cli output.

```
D:\zaid\zaid\codes\asasas> java -jar checkstyle-10.23.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] D:\zaid\zaid\codes\asasas\Test.java:5:9: Distance between variable 'a' declaration and its first usage is 3, but allowed 1.  Consider making that variable final if you still need to store its value in advance (before method calls that might have side effects on the original value). [VariableDeclarationUsageDistance]
Audit done.
Checkstyle ends with 1 errors
```


this ss shown below is the google style guideline 

![Image](https://github.com/user-attachments/assets/37bf32d5-a55f-466f-bd50-7b7e944fa76f)

and the error received here is directly affecting the guideline and the implementation of the check.

**Comment by romani** [comment]
There is some incomplete update at https://github.com/checkstyle/checkstyle/pull/17526 that can be reuses.
Everyone is welcome to send PR

**Comment by vivek-0509** [comment]
I am on it

**Comment by romani** [comment]
fix is merged
